### PR TITLE
AIDA-813: Make invalid tests confirm violation counts

### PR DESCRIPTION
**Pull request by ecurley-nextcentury**
Modify OntologyGeneration to produce ShaclShapes.java.
Add ShaclShapes.java.
Modify ExamplesAndValidationTest, NistExamplesAndValidationTest, and NistTA3ExamplesAndValidationTest to provided expected violations.
Add TestUtils.expectedCounts and TestUtils.expect to specify what violations to expect.
Modify TestUtils.testInvalid to require that expectedCounts be populated.

### CODE REVIEW

File: src/main/java/com/ncc/aif/ont2javagen/OntologyGeneration.java [hunk 1]
```diff
@@ -92,15 +98,63 @@ public static void main(String[] args) {
...
+            out.println("package com.ncc.aif;\n" +
+                    "\nimport org.apache.jena.rdf.model.Resource;\n" +
+                    "import org.apache.jena.rdf.model.ResourceFactory;\n" +
+                    "\n// WARNING. This is a Generated File.  Please do not edit.\n" +
+                    "// This class contains variables generated from SHACL files using the OntologyGeneration Class\n" +
+                    "// Please refer to the README at src/main/java/com/ncc/aif/ont2javagen for more information\n" +
+                    "// Last Generated On: " + new SimpleDateFormat("MM/dd/yyy HH:mm:ss").format(new Date()));
+            out.println("public final class " + className + " {");
+
+            String indent = "    ";
```
[the remark this answers is not in the document]
Comment (ecurley-nextcentury) [reply src/main/java/com/ncc/aif/ont2javagen/OntologyGeneration.java]: Tab shows up as different sizes in different environments. Four spaces always shows up the same. It makes the source file ~2 bytes bigger, but I like the consistency. I'm also pretty sure we use spaces in our other source files.

File: src/main/java/com/ncc/aif/ont2javagen/OntologyGeneration.java [hunk 2]
```diff
@@ -83,7 +88,8 @@ private void setResources(String uri, String resourceName) {
     public static void main(String[] args) {
 
         if (args.length == 0) {
-            String root = "src/main/resources/com/ncc/aif/ontologies/";
+            String aifRoot = "src/main/resources/com/ncc/aif/";
+            String ontRoot = "ontologies/";
```
[the remark this answers is not in the document]
Comment (ecurley-nextcentury) [reply src/main/java/com/ncc/aif/ont2javagen/OntologyGeneration.java]: Well, that's embarrassing. Thanks for catching this.

File: src/test/java/com/ncc/aif/ExamplesAndValidationTest.java [hunk 3]
```diff
@@ -1085,13 +1087,16 @@ void eventMissingType() {
             // having multiple type assertions in case of uncertainty is ok, but there must always
             // be at least one type assertion
             makeEvent(model, utils.getEventUri(), system);
+            utils.expect(ShaclShapes.EventRelationShape, SH.SPARQLConstraintComponent, null);
```
[the remark this answers is not in the document]
Comment (ecurley-nextcentury) [reply src/test/java/com/ncc/aif/ExamplesAndValidationTest.java]: Personally, I believe that even if any of the performance enhancements ever make it into master, we're still going to be dealing with a large body of data to validate, so I think this still holds.

In addition, as Events and Relations share all the same restrictions, they should be a single shape. 

If the name is confusing, we should modify it to be less so. Any suggestions?

[the remark this answers is not in the document]
Comment (sharknc) [reply src/test/java/com/ncc/aif/ExamplesAndValidationTest.java]: Maybe `EventOrRelationShape`?

**Comment by ecurley-nextcentury** [comment]
> * `invalidNumericValueAsString()` in `NistExamplesAndValidationTest.java`

This test is disabled so I didn't address it. It fails when run, though, so it can be addressed when we enable it

> * `invalid()` under `KEsInModelMustBeReferencedByHypothesis`  in `NistTA3ExamplesAndValidationTest.java`

This does actually use the TestUtils.expect method. Unfortunately, I put it at the top of the method instead of just before the testInvalid method. I'll move it.

File: src/test/java/com/ncc/aif/ExamplesAndValidationTest.java (hunk 3 above)
[the remark this answers is not in the document]
Comment (ecurley-nextcentury) [reply src/test/java/com/ncc/aif/ExamplesAndValidationTest.java]: Sounds good to me. I'll write up a ticket for this as I've been told that we shouldn't make code changes that aren't described in the ticket the branch is associated with.

[the remark this answers is not in the document]
Comment (ecurley-nextcentury) [reply src/test/java/com/ncc/aif/ExamplesAndValidationTest.java]: Added https://nextcentury.atlassian.net/browse/AIDA-839 to address this

File: src/main/java/com/ncc/aif/ont2javagen/OntologyGeneration.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (ecurley-nextcentury) [reply src/main/java/com/ncc/aif/ont2javagen/OntologyGeneration.java]: fixed

**Pull request merged**